provider/oci/oci.go

```diff
@@ -113,27 +115,61 @@ func (p *OCIProvider) zones(ctx context.Context) (map[string]dns.ZoneSummary, er
 
 	log.Debugf("Matching zones against domain filters: %v", p.domainFilter.Filters)
 	var page *string
-	for {
-		resp, err := p.client.ListZones(ctx, dns.ListZonesRequest{
-			CompartmentId: &p.cfg.CompartmentID,
-			ZoneType:      dns.ListZonesZoneTypePrimary,
-			Page:          page,
-		})
-		if err != nil {
-			return nil, errors.Wrapf(err, "listing zones in %q", p.cfg.CompartmentID)
-		}
 
-		for _, zone := range resp.Items {
-			if p.domainFilter.Match(*zone.Name) && p.zoneIDFilter.Match(*zone.Id) {
-				zones[*zone.Name] = zone
-				log.Debugf("Matched %q (%q)", *zone.Name, *zone.Id)
-			} else {
-				log.Debugf("Filtered %q (%q)", *zone.Name, *zone.Id)
+	// set zone scope based on configuration flag
+	scope := dns.ListZonesScopeEnum(p.zoneScope)
+	if scope == "" {
```

Comment:
Instead of duplicating the code below for the empty scope, you could build a slice of scopes here, containing just the value the user provided, or both scopes if it's an empty string. Then just always loop through that slice as you are below.